src/shed/__init__.py

```diff
@@ -146,34 +232,35 @@ def shed(
     if refactor and not is_pyi:
         source_code = _run_codemods(source_code, min_version=min_version)
 
-    def run_isort() -> str:
-        try:
-            return isort.code(
-                source_code,
-                known_first_party=first_party_imports,
-                known_local_folder={"tests"},
-                profile="black",
-                combine_as_imports=True,
-            )
-        except FileSkipComment:
-            return source_code
-
-    # Autoflake cannot always correctly resolve imports it can remove until
-    # they have been properly formatted, so we have to run isort first so that
-    # it gets well formatted imports.
-    pre_autoflake = source_code = run_isort()
-
-    source_code = autoflake.fix_code(
-        source_code,
-        expand_star_imports=True,
-        remove_all_unused_imports=_remove_unused_imports,
-    )
+    def run_ruff() -> str:
+        # I; isort; sort imports
+        # PIE790; unnecessary-placeholder; unnecessary pass/... statement
+        # F841; unused-variable
+        select = ",".join(_RUFF_RULES)
+        if _remove_unused_imports:
+            select += ",F401"
+        sub = subprocess.run(
+            [
+                "ruff",
+                "check",
+                f"--select={select}",
+                "--fix-only",
+                f"--target-version=py3{min_version[1]}",
+                "--isolated",  # ignore configuration files
+                "--exit-zero",  # Exit with 0, even upon detecting lint violations.
+                "--config=lint.isort.combine-as-imports=true",
+                f"--config=lint.isort.known-first-party={list(first_party_imports)}",
+                f"--config=lint.extend-safe-fixes={list(_RUFF_EXTEND_SAFE_FIXES)}",
+                "-",  # pass code on stdin
+            ],
+            input=source_code,
+            encoding="utf-8",
+            check=True,
+            capture_output=True,
+        )
+        return sub.stdout
 
-    # Until https://github.com/PyCQA/autoflake/issues/229 is fixed, autoflake
-    # may reorder imports in a way that is incompatible with isort's ordering,
-    # so we may need to re-sort its output.
-    if source_code != pre_autoflake:
-        source_code = run_isort()
+    source_code = run_ruff()
```

Comment:
Since we're only running `ruff` once, we can just inline the function definition and save a bit of time 🙂